api/v1alpha1/clientprofile_types.go

```diff
@@ -23,51 +23,60 @@ import (
 
 // CephFsConfigSpec defines the desired CephFs configuration
 type CephFsConfigSpec struct {
+	//+kubebuilder:validation:Optional
 	SubVolumeGroup string `json:"subVolumeGroup,omitempty"`
 }
 
 // RbdConfigSpec defines the desired RBD configuration
 type RbdConfigSpec struct {
+	//+kubebuilder:validation:Optional
 	RadosNamespace string `json:"radosNamespace,omitempty"`
 }
 
 // NfsConfigSpec cdefines the desired NFS configuration
 type NfsConfigSpec struct {
 }
 
-// ConfigSpec defines the desired state of Config
-type ConfigSpec struct {
+// ClientProfileSpec defines the desired state of Config
+type ClientProfileSpec struct {
+	//+kubebuilder:validation:Required
 	CephClusterRef corev1.LocalObjectReference `json:"cephClusterRef"`
-	CephFs         *CephFsConfigSpec           `json:"cephFs,omitempty"`
-	Rbd            *RbdConfigSpec              `json:"rbd,omitempty"`
-	Nfs            *NfsConfigSpec              `json:"nfs,omitempty"`
+
+	//+kubebuilder:validation:Optional
+	CephFs *CephFsConfigSpec `json:"cephFs,omitempty"`
+
+	//+kubebuilder:validation:Optional
+	Rbd *RbdConfigSpec `json:"rbd,omitempty"`
+
+	//+kubebuilder:validation:Optional
+	Nfs *NfsConfigSpec `json:"nfs,omitempty"`
 }
 
-// ConfigStatus defines the observed state of Config
-type ConfigStatus struct {
+// ClientProfileStatus defines the observed state of Config
+type ClientProfileStatus struct {
 }
 
 //+kubebuilder:object:root=true
 //+kubebuilder:subresource:status
 
-// Config is the Schema for the configs API
-type Config struct {
+// ClientProfile is the Schema for the configs API
+type ClientProfile struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   ConfigSpec   `json:"spec,omitempty"`
-	Status ConfigStatus `json:"status,omitempty"`
+	Spec   ClientProfileSpec   `json:"spec,omitempty"`
+	Status ClientProfileStatus `json:"status,omitempty"`
 }
 
 //+kubebuilder:object:root=true
 
-// ConfigList contains a list of Config
-type ConfigList struct {
+// ClientProfileList contains a list of Config
```

Comment:
// ClientProfileList contains a list of CephFS, RBD, and NFS configuration.

Reply:
done with some changes

Reply:
looks good.